kotlinx-coroutines-core/jvm/src/Executors.kt

```diff
@@ -107,6 +107,13 @@ public fun Executor.asCoroutineDispatcher(): CoroutineDispatcher =
 public fun CoroutineDispatcher.asExecutor(): Executor =
     (this as? ExecutorCoroutineDispatcher)?.executor ?: DispatcherExecutor(this)
 
+/** Executes the [Runnable] and suspends until the execution completes. */
+public inline suspend operator fun Executor.invoke(runnable: Runnable) {
+    withContext(asCoroutineDispatcher()) {
```

Comment:
isn't this going to create a new dispatcher for each invocation?

Reply:
Although I closed this PR I'm curious, is that an issue?

Reply:
It's always better to avoid running code several times when one time suffices. `asCoroutineDispatcher` is a small memory allocation, followed by a reflective method call if the `Executor` is a `ScheduledThreadPoolExecutor`. When it comes to reflective calls, your mileage may vary w.r.t. how slow they are, but in any case, I doubt that this would show up in a profiler in the overwhelming majority of cases.